server/sonar-server/src/main/java/org/sonar/server/platform/ws/RestartAction.java

```diff
@@ -55,14 +67,35 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) {
-    if (!settings.getBoolean("sonar.web.dev")) {
-      throw new ForbiddenException("Webservice available only in dev mode");
+    if (settings.getBoolean("sonar.web.dev")) {
+      LOGGER.info("Fast restarting WebServer...");
+      platform.restart();
+      LOGGER.info("WebServer restarted");
+      response.noContent();
+    } else {
```

Comment:
Contrary to dev mode, response is not written here. `response.noContent()` can't be executed ?